### PR TITLE
build: add technology repo release workflows

### PR DESCRIPTION
**Pull request by ndr-brt**
## What this PR changes/adds

Adds workflows to be called by the `Technology-` repos on release.
ref. https://github.com/eclipse-edc/docs/pull/189

## Why it does that

_Briefly state why the change was necessary._

## Further notes

- removed the `release-component` workflow, that's has not been used in the last months 

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: .github/workflows/technology-prepare-release.yml
Comment (paullatzelsperger) [inline .github/workflows/technology-prepare-release.yml:53]: adding spaces to the regex is probably safer. same for the project version above.
```suggestion
          sed -i 's#^edc\s*=\s*.*#edc = "${{ inputs.version }}"#g' gradle/libs.versions.toml
```

File: .github/workflows/technology-prepare-release.yml
Comment (paullatzelsperger) [inline .github/workflows/technology-prepare-release.yml:54]: Technically, we'd also have to update the version in the `DEPENDENCIES` file, otherwise we'd be releasing artefacts with wrong dependencies listed.

Since we have to do that, how about simply parsing the "old" version from `gradle.properties` like so:
```bash
existingVersion=$(grep "version" gradle.properties  | awk -F= '{print $2}')
```

and then all files could be `sed`-ed in one go:
```bash
grep -rlz "$existingVersion" . --exclude=\*.{sh,bin} | xargs sed -i "s/$existingVersion/${{ inputs.version }}/g"
```

File: .github/workflows/technology-prepare-release.yml
Comment (paullatzelsperger) [inline .github/workflows/technology-prepare-release.yml:63]: `-f` shouldn't be necessary, in fact it is superfluous, because if the branch `prepare/....` had existed before, `git checkout -b` would have failed.

File: .github/workflows/technology-release.yml
Comment (paullatzelsperger) [inline .github/workflows/technology-release.yml:63]: here, `-f` is potentially harmful. What would be the use case of needing to overwrite tags?

**Review by paullatzelsperger — COMMENTED** [review]
cool stuff! Am I correct in assuming that every tech repo will have to have a `prepare-release.yaml` and `publish-release.yaml` workflow, which are triggered manually, or upon PR merging respectively, that in turn call up these WF's here?

File: .github/workflows/technology-prepare-release.yml
Comment (ndr-brt) [reply .github/workflows/technology-prepare-release.yml:54]: the nice thing about this new process is that the `preparation` phase will create a PR that will need to be merged manually, so in that one all the checks should run, dependencies one included, at that point the committer will need to fix everything (and eventually, for a bugfix, cherry-picking the fix commits) before merging it and trigger the release

File: .github/workflows/technology-prepare-release.yml
Comment (ndr-brt) [reply .github/workflows/technology-prepare-release.yml:63]: I added it in the case the preparation failed for some reasons and there's the need to re-run it again

File: .github/workflows/technology-release.yml
Comment (ndr-brt) [reply .github/workflows/technology-release.yml:63]: same here, in case of eventual failure, but, true, at this point artifacts would have been already published so there's no real need for this `force`

**Comment by ndr-brt** [comment]
> cool stuff! Am I correct in assuming that every tech repo will have to have a `prepare-release.yaml` and `publish-release.yaml` workflow, which are triggered manually, or upon PR merging respectively, that in turn call up these WF's here?

yes, once this is merged I will create the workflows in the repos

File: .github/workflows/technology-prepare-release.yml
Comment (paullatzelsperger) [reply .github/workflows/technology-prepare-release.yml]: that's fine, too, but then the checkout should be:
```bash
git checkout ... || git checkout -b ...
```

**Review by paullatzelsperger — APPROVED** [review]

**Comment by paullatzelsperger** [comment]
> EDIT: maybe we could have the proper release automatically triggering the worflow after merge while the bugfix could be manual to permit cherry-picks and backports to happen separately.

this should be easily doable by restricting the trigger to the `release/` branch prefix: 
```yaml
on:
  pull_request:
    branches:
      - releases
    types:
      - closed
  workflow_dispatch:
```

**Pull request merged**
